### PR TITLE
🎉 ported over fusionannnotator

### PR DESCRIPTION
**Pull request by migbro**
## Description

Ported over Dockerfile used to generate a prod docker image from https://github.com/kgaonkar6/tools/blob/master/FusionAnnotator/Dockerfile (thanks @sickler-alex !). Updated ubuntu image and also added pigz to speed up untar of large reference tar ball.

Fixes # https://github.com/d3b-center/bixu-tracker/issues/1748

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Run as part of annoFuse wf test: https://cavatica.sbgenomics.com/u/kids-first-drc/kids-first-drc-rnaseq-workflow/tasks/c7c03721-5081-4620-ab19-5510a1f476ab/
- [ ] Not a direct output, but downstream outputs identical [before](https://cavatica.sbgenomics.com/u/kids-first-drc/kids-first-drc-rnaseq-workflow/files/63b7081d9668d812ed3016c6/) and [after](https://cavatica.sbgenomics.com/u/kids-first-drc/kids-first-drc-rnaseq-workflow/files/63c706db5ec27133d2f2d00d/)
**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

### CODE REVIEW

**Review by dmiller15 — APPROVED** [review]
I know this is just a port with little update but I think for downloading and untarring gz files we should be piping curl into tar. For example:
```
RUN wget "https://cpan.metacpan.org/authors/id/M/ML/MLEHMANN/common-sense-3.74.tar.gz"
RUN tar -zxvf Canary-Stability-2013.tar.gz &&...
RUN rm *.tar.gz
```
Can be simplified to:
```
RUN curl -SL "https://cpan.metacpan.org/authors/id/M/ML/MLEHMANN/common-sense-3.74.tar.gz" | tar -zx &&...
```

**Comment by migbro** [comment]
> I know this is just a port with little update but I think for downloading and untarring gz files we should be piping curl into tar. For example:
> 
> ```
> RUN wget "https://cpan.metacpan.org/authors/id/M/ML/MLEHMANN/common-sense-3.74.tar.gz"
> RUN tar -zxvf Canary-Stability-2013.tar.gz &&...
> RUN rm *.tar.gz
> ```
> 
> Can be simplified to:
> 
> ```
> RUN curl -SL "https://cpan.metacpan.org/authors/id/M/ML/MLEHMANN/common-sense-3.74.tar.gz" | tar -zx &&...
> ```

Nice! I like that suggestion and honestly hadn't though of it

File: FusionAnnotator/0.1.1/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+RUN curl -SL "https://cpan.metacpan.org/authors/id/M/ML/MLEHMANN/JSON-XS-4.02.tar.gz" | tar -zx && \
+    cd JSON-XS-4.02 && \
+    perl Makefile.PL && \
+    make && \
+    make install
+
+RUN apt remove -y build-essential curl git && apt autoclean -y
+WORKDIR /opt/FusionAnnotator/
+ENV FUSION_ANNOTATOR /opt/FusionAnnotator/FusionAnnotator
```
Comment (sickler-alex) [inline FusionAnnotator/0.1.1/Dockerfile:62]: Do we want to include adding the docker file like we do with our other dockers or do we want to just directly copy this?

Comment (migbro) [reply FusionAnnotator/0.1.1/Dockerfile:62]: Good catch! I missed that

**Pull request merged**